### ISSUE TITLE
Expand the use of context dependency injection

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/phoenicis#830]
Currently we're using a lot of parameter passing. This makes writing and managing our classes quite cumbersome. Therefore I would suggest to use more context and dependency injection if possible.
As a first step I would suggest to change our controllers to be managed by spring.
This would enable us to use `@Autowired` instead of parameter passing to get our parameters inside the controllers.
What do you think?

**Comment by qparis** [comment]
I don't like using @Autowired annotation for several reasons : 
- When you use @Autowired, you are encourage a dependency in your class without thinking about the dependencies of your modules. Basicallly, you could autowire the universe everywhere
- Using constructors injection is a perfectly acceptable pattern from a Spring point of view

**Comment by qparis** [comment]
If we have to much dependencies, it may indicate that the component has too much responsibility and we may need to create an extra class.
This information is very valuable for us and we cannot see it clearly when using @Autowired

**Comment by madoar** [comment]
Normally I would fully agree with you, but take a look at the `MainController`. It has the task to more or less plug the different controllers together in ~50 lines of code, but also requires 8 parameters in its constructor. I would like to get rid of such classes.
If we can't use CDI here, then we should maybe think about our controller class hierarchy?

**Comment by qparis** [comment]
For this particular case, I agree with you. But I would rather not to use auto injection in other places

**Comment by madoar** [comment]
Currently I'm only thinking about changing our controllers to use CDI. Other classes should be more a case for responsibility splitting.

**Comment by plata** [comment]
@madoar @qparis did you reach an agreement if/what we should do about this? If so, please add distinct tasks for the actions that have to be taken. If not, can we just close this issue?

**Comment by madoar** [comment]
The idea is to use spring in the Controller classes.
This would reduce the need for our many argument controller constructors.
To do this we basically only need to add all the controller parameters as class fields annotated with  `@Autowired` and remove the many argument constructors.
In a next step (and separate PR) I would like to test if and how well this same approach would work with our View classes, i.e the tab instances, which each exist only once.

**Comment by plata** [comment]
Can we take a decision here?

**Comment by madoar** [comment]
I'm not sure this is feasible anymore, because I'm currently trying to remove our controller classes.
As far as I know the only easy approach to use dependency injection with javafx is via FXML, which I don't want to use (see #1763 for the reason).

**Comment by plata** [comment]
Ok. Then I will close this now.
